File: 30daysOFcode_Go/day 18.go
package main
import "fmt"

func main() {
	var s string
	fmt.Scan(&s)
	obj := &Solution{}
	for _, c := range s {
		obj.pushCharacter(c)
		obj.enqueueCharacter(c)
	}

	isPalindrome := true
	for i := 0; i < len(s)/2; i++ {
		if obj.popCharacter() != obj.dequeueCharacter() {
			isPalindrome = false
			break
		}
	}

	if isPalindrome {
		fmt.Println("The word, "+s+", is a palindrome.")
	} else {
		fmt.Println("The word, "+s+", is not a palindrome.")
	}
}

type Solution struct {
	queue []rune
	stack []rune
}

func (s *Solution) pushCharacter(c rune) {
	s.stack = append(s.stack, c)
}
func (s *Solution) enqueueCharacter(c rune) {
	s.queue = append(s.queue, c)
}
func (s *Solution) popCharacter() (rune) {
	c := s.stack[len(s.stack) - 1]
	s.stack = s.stack[:len(s.stack) - 1]
	return c
}
func (s *Solution) dequeueCharacter() (rune) {
	c := s.queue[0]
	s.queue = s.queue[1:]
	fmt.Println(c)
	return c
}

File: 30daysOFcode_Php/day18.php
<?php

class Solution
{
    private $stack = [];
    private $queue = [];

    public function pushCharacter($stack)
    {
        $this->stack[] = $stack;
        array_push($this->stack, $stack);
    }

    public function enqueueCharacter($queue)
    {
        $this->queue[] = $queue;
        array_push($this->queue, $queue);
    }

    public function popCharacter()
    {
        return array_pop($this->stack);
    }

    public function dequeueCharacter()
    {
        return array_shift($this->queue);
    }
}

// read the string s
$s = fgets(STDIN);

// create the Solution class object p
$obj = new Solution();

$len = strlen($s);
$isPalindrome = true;

// push/enqueue all the characters of string s to stack
for ($i = 0; $i < $len; $i++) {
    $obj->pushCharacter($s{$i});
    $obj->enqueueCharacter($s{$i});
}

/*
pop the top character from stack
dequeue the first character from queue
compare both the characters
*/
for ($i = 0; $i < $len / 2; $i++) {
    if ($obj->popCharacter() != $obj->dequeueCharacter()) {
        $isPalindrome = false;

        break;
    }
}

//finally print whether string s is palindrome or not.
if ($isPalindrome) {
    echo 'The word, ' . $s . ', is a palindrome.';
} else {
    echo 'The word, ' . $s . ', is not a palindrome.';
}
